### PR TITLE
fix: Add possibility to provide custom options when publishing

### PR DESCRIPTION
**Pull request by stambata**
Add possibility to provide custom options when publishing to an exchange

### CODE REVIEW

**Review by stoberov — APPROVED** [review]

File: produce/index.js [hunk 1]
```diff
@@ -18,15 +18,39 @@ module.exports = ({utBus, registerErrors, ...rest}) => class AmqpProducer extend
...
+
+        const [exchange, routingKey] = method.split('.').slice(1, 3);
+
+        const {type, opts} = this.config.exchange[exchange];
+
+        let payload, options;
+
+        if (msg.payload) {
+            payload = msg.payload;
+            options = {...publishOptions, ...msg.options};
```
Comment (peterdenev) [inline produce/index.js:37]: Here you can override the exchange and routing key if you pass them to the msg. This way port can use routing keys with dots "."

**Review by peterdenev — COMMENTED** [review]
what about routing key with dots

File: produce/index.js (hunk 1 above)
Comment (stambata) [reply produce/index.js:37]: Noted. Please see the changes applied in the next commit.

Comment (stoberov) [reply produce/index.js:37]: Looks like it's going to work splendidly :)

**Review by stoberov — APPROVED** [review]

File: produce/index.js [hunk 2]
```diff
@@ -29,20 +30,22 @@ module.exports = ({utBus, registerErrors, ...rest}) => class AmqpProducer extend
...
         if (msg.payload) {
             payload = msg.payload;
             options = {...publishOptions, ...msg.options};
+            routingKey = msg.routingKey;
         } else {
             payload = msg;
             options = publishOptions;
         }
 
+        const [exchange, ...routingKeyTokens] = method.split('.').slice(1);
```
Comment (peterdenev) [inline produce/index.js:44]: I hope this will not affect pre/after hooks (request.send, response.receive)?

Comment (stambata) [reply produce/index.js:44]: Nope. this is the exec method which gets called with the original method only. send/receive conversions are separate.

**Pull request merged**